gui/layers/_markers_layer.py

```diff
@@ -67,17 +67,20 @@ def __init__(
         visual = MarkersNode()
         super().__init__(visual)
 
+        # Block any refreshes during parameter setting
+        self._freeze = True
```

Comment:
This should probably be a context manager: you don't want the freeze to persist if there is an error below.

```python
with self.freeze_refresh():
   # do the things
```

Reply:
I have now changed this

Reply:
I have fixed this